src/zepben/auth/client/zepben_token_fetcher.py

```diff
@@ -6,51 +6,63 @@
 import warnings
 from datetime import datetime
 from typing import Optional, Union, Callable, Dict
+
 import jwt
 import requests
 from dataclassy import dataclass
 from urllib3.exceptions import InsecureRequestWarning
-
-from zepben.auth.client.util import construct_url
 from zepben.auth.common.auth_exception import AuthException
 from zepben.auth.common.auth_method import AuthMethod
+from zepben.auth.common.auth_provider_config import AuthProviderConfig, create_auth_provider_config, fetch_provider_details
 
 
 __all__ = ["ZepbenTokenFetcher", "create_token_fetcher", "get_token_fetcher", "create_token_fetcher_managed_identity"]
 
 
-def _fetch_token_generator(is_azure: bool, use_identity: bool, identity_url: Optional[str] = None) -> Callable[[Dict, Dict, str, str, str, bool, bool], requests.Response]:
-    def _post_azure(refresh_request_data: Dict, token_request_data: Dict, issuer_protocol: str, issuer_domain: str, token_path: str, refresh: bool, verify: bool):
-        # Azure requires a scope of the audience + /.default. We strip the / just in case so we don't end up with 2 slashes.
-        aud = refresh_request_data["audience"].rstrip('/') + "/.default"
-        refresh_request_data["scope"] = aud
+def _fetch_token_generator(is_entraid: bool, use_identity: bool, identity_url: Optional[str] = None) -> Callable[
+    [Dict, Dict, str, str, str, bool, bool], requests.Response]:
+    def post(
+        refresh_request_data: Dict,
+        token_request_data: Dict,
+        token_endpoint: str,
+        refresh: bool,
+        verify: bool
+    ):
+
+        # EntraID requires a scope of the audience + /.default. We strip the / just in case so we don't end up with 2 slashes.
+        refresh_request_data["scope"] = refresh_request_data["audience"] + "/.default"
         # Also requires grant type client credentials for access tokens
         token_request_data.update({
             'grant_type': 'client_credentials',
```

Comment:
These flows don't usually mix, however, if you connect with username/password that comes _after_ this step is finished (~line 245), so it will be overwritten to use an appropriate method (afaik though, EntraID token don't support user/password, but I'm probably wrong).